<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Controller</class>
 <widget class="QMainWindow" name="Controller">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>989</width>
    <height>664</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Controller</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QTabWidget" name="tab_manager">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="minimumSize">
       <size>
        <width>100</width>
        <height>0</height>
       </size>
      </property>
      <property name="locale">
       <locale language="English" country="Australia"/>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="Robot1">
       <attribute name="title">
        <string>Robot1</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <widget class="QGroupBox" name="Camera">
          <property name="sizePolicy">
           <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="title">
           <string>Camera</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_7">
           <item>
            <widget class="CQtOpenCVViewerGl" name="openCVviewer" native="true"/>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="Robot2">
       <attribute name="title">
        <string>Robot2</string>
       </attribute>
       <widget class="QGroupBox" name="Camera_2">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>10</y>
          <width>453</width>
          <height>522</height>
         </rect>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="title">
         <string>Camera</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_8">
         <item>
          <widget class="CQtOpenCVViewerGl" name="openCVviewer_2" native="true"/>
         </item>
        </layout>
       </widget>
      </widget>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QDockWidget" name="dock_status">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>372</width>
        <height>571</height>
       </size>
      </property>
      <property name="allowedAreas">
       <set>Qt::RightDockWidgetArea</set>
      </property>
      <property name="windowTitle">
       <string>Command Panel</string>
      </property>
      <widget class="QWidget" name="dockWidgetContents_4">
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QFrame" name="frame_5">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <widget class="QGroupBox" name="groupBox_4">
             <property name="title">
              <string>Robot Control</string>
             </property>
             <widget class="QPushButton" name="go">
              <property name="geometry">
               <rect>
                <x>170</x>
                <y>40</y>
                <width>89</width>
                <height>25</height>
               </rect>
              </property>
              <property name="text">
               <string>Go</string>
              </property>
             </widget>
             <widget class="QPushButton" name="back">
              <property name="geometry">
               <rect>
                <x>170</x>
                <y>100</y>
                <width>89</width>
                <height>25</height>
               </rect>
              </property>
              <property name="text">
               <string>Back</string>
              </property>
             </widget>
             <widget class="QPushButton" name="left">
              <property name="geometry">
               <rect>
                <x>76</x>
                <y>70</y>
                <width>89</width>
                <height>25</height>
               </rect>
              </property>
              <property name="text">
               <string>Left Turn</string>
              </property>
             </widget>
             <widget class="QPushButton" name="right">
              <property name="geometry">
               <rect>
                <x>264</x>
                <y>70</y>
                <width>89</width>
                <height>25</height>
               </rect>
              </property>
              <property name="text">
               <string>Right Turn</string>
              </property>
             </widget>
             <widget class="QPushButton" name="uturn">
              <property name="geometry">
               <rect>
                <x>170</x>
                <y>70</y>
                <width>89</width>
                <height>25</height>
               </rect>
              </property>
              <property name="text">
               <string>U Turn</string>
              </property>
             </widget>
             <widget class="QSplitter" name="splitter">
              <property name="geometry">
               <rect>
                <x>10</x>
                <y>132</y>
                <width>321</width>
                <height>31</height>
               </rect>
              </property>
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <widget class="QLabel" name="label_4">
               <property name="text">
                <string>Operation Mode</string>
               </property>
              </widget>
              <widget class="QRadioButton" name="auto_2">
               <property name="text">
                <string>Autonomous</string>
               </property>
              </widget>
              <widget class="QRadioButton" name="manual">
               <property name="text">
                <string>Manual</string>
               </property>
              </widget>
             </widget>
             <widget class="QLabel" name="status_red">
              <property name="geometry">
               <rect>
                <x>390</x>
                <y>2</y>
                <width>31</width>
                <height>17</height>
               </rect>
              </property>
              <property name="text">
               <string/>
              </property>
              <property name="pixmap">
               <pixmap>assets/bullet_red.png</pixmap>
              </property>
             </widget>
             <widget class="QLabel" name="status_green">
              <property name="geometry">
               <rect>
                <x>367</x>
                <y>2</y>
                <width>31</width>
                <height>17</height>
               </rect>
              </property>
              <property name="text">
               <string/>
              </property>
              <property name="pixmap">
               <pixmap>assets/bullet_green.png</pixmap>
              </property>
             </widget>
             <widget class="QLabel" name="status_yellow">
              <property name="geometry">
               <rect>
                <x>412</x>
                <y>2</y>
                <width>31</width>
                <height>17</height>
               </rect>
              </property>
              <property name="text">
               <string/>
              </property>
              <property name="pixmap">
               <pixmap>assets/bullet_yellow.png</pixmap>
              </property>
             </widget>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QFrame" name="frame_6">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="layoutDirection">
           <enum>Qt::LeftToRight</enum>
          </property>
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <property name="lineWidth">
           <number>1</number>
          </property>
          <widget class="QGroupBox" name="groupBox_5">
           <property name="geometry">
            <rect>
             <x>9</x>
             <y>9</y>
             <width>441</width>
             <height>161</height>
            </rect>
           </property>
           <property name="title">
            <string>Camera Control</string>
           </property>
           <widget class="QCheckBox" name="start">
            <property name="geometry">
             <rect>
              <x>10</x>
              <y>30</y>
              <width>124</width>
              <height>23</height>
             </rect>
            </property>
            <property name="text">
             <string>On/Off</string>
            </property>
           </widget>
           <widget class="QSlider" name="horizontalSlider">
            <property name="geometry">
             <rect>
              <x>260</x>
              <y>90</y>
              <width>84</width>
              <height>30</height>
             </rect>
            </property>
            <property name="minimum">
             <number>50</number>
            </property>
            <property name="maximum">
             <number>250</number>
            </property>
            <property name="value">
             <number>150</number>
            </property>
            <property name="sliderPosition">
             <number>150</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="invertedAppearance">
             <bool>false</bool>
            </property>
            <property name="invertedControls">
             <bool>false</bool>
            </property>
            <property name="tickPosition">
             <enum>QSlider::TicksAbove</enum>
            </property>
            <property name="tickInterval">
             <number>50</number>
            </property>
           </widget>
           <widget class="QSlider" name="verticalSlider">
            <property name="geometry">
             <rect>
              <x>210</x>
              <y>60</y>
              <width>30</width>
              <height>84</height>
             </rect>
            </property>
            <property name="minimum">
             <number>80</number>
            </property>
            <property name="maximum">
             <number>220</number>
            </property>
            <property name="value">
             <number>150</number>
            </property>
            <property name="sliderPosition">
             <number>150</number>
            </property>
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="tickPosition">
             <enum>QSlider::TicksAbove</enum>
            </property>
            <property name="tickInterval">
             <number>50</number>
            </property>
           </widget>
           <widget class="QLabel" name="label_5">
            <property name="geometry">
             <rect>
              <x>230</x>
              <y>30</y>
              <width>71</width>
              <height>21</height>
             </rect>
            </property>
            <property name="layoutDirection">
             <enum>Qt::LeftToRight</enum>
            </property>
            <property name="text">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Pan/Tilt&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
           </widget>
          </widget>
         </widget>
        </item>
        <item>
         <widget class="QFrame" name="frame">
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <widget class="QGroupBox" name="groupBox_6">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>10</y>
             <width>441</width>
             <height>161</height>
            </rect>
           </property>
           <property name="title">
            <string>Diagnostic Control</string>
           </property>
           <widget class="QLabel" name="label">
            <property name="geometry">
             <rect>
              <x>13</x>
              <y>105</y>
              <width>91</width>
              <height>17</height>
             </rect>
            </property>
            <property name="text">
             <string>Command</string>
            </property>
           </widget>
           <widget class="QWidget" name="layoutWidget">
            <property name="geometry">
             <rect>
              <x>13</x>
              <y>30</y>
              <width>421</width>
              <height>72</height>
             </rect>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_3">
             <item>
              <widget class="QPlainTextEdit" name="diag">
               <property name="verticalScrollBarPolicy">
                <enum>Qt::ScrollBarAlwaysOn</enum>
               </property>
              </widget>
             </item>
             <item>
              <layout class="QVBoxLayout" name="verticalLayout">
               <item>
                <widget class="QPushButton" name="log">
                 <property name="text">
                  <string>Log</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="save">
                 <property name="text">
                  <string>Save</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
           <widget class="QWidget" name="layoutWidget">
            <property name="geometry">
             <rect>
              <x>13</x>
              <y>127</y>
              <width>421</width>
              <height>27</height>
             </rect>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <item>
              <widget class="QLineEdit" name="command"/>
             </item>
             <item>
              <widget class="QPushButton" name="send">
               <property name="text">
                <string>Send</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>989</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuMenu">
    <property name="title">
     <string>Menu</string>
    </property>
    <addaction name="actionQuit"/>
   </widget>
   <addaction name="menuMenu"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
  </action>
  <action name="status_green_2">
   <property name="text">
    <string>connected</string>
   </property>
  </action>
  <action name="status_red_2">
   <property name="text">
    <string>disconnected</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>CQtOpenCVViewerGl</class>
   <extends>QWidget</extends>
   <header location="global">cqtopencvviewergl.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>actionQuit</sender>
   <signal>triggered(bool)</signal>
   <receiver>Controller</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>494</x>
     <y>331</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
